modules/consensus/accept_test.go

```diff
@@ -701,3 +701,64 @@ func TestTaxHardfork(t *testing.T) {
 		t.Fatal("siafund pool was not increased correctly")
 	}
 }
+
+// mockGateway implements modules.Gateway to mock the Broadcast method.
+type mockGateway struct {
+	modules.Gateway
+	numBroadcasts int
+}
+
+func (g *mockGateway) Broadcast(string, interface{}) {
+	g.numBroadcasts++
+	return
+}
+
+// TestAcceptBlockBroadcasts tests that AcceptBlock broadcasts valid blocks and
+// that managedAcceptBlock does not.
+func TestAcceptBlockBroadcasts(t *testing.T) {
+	cst, err := blankConsensusSetTester("TestAcceptBlockBroadcasts")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst.Close()
+	mg := &mockGateway{Gateway: cst.cs.gateway}
+	cst.cs.gateway = mg
+
+	// Test that Broadcast is called for valid blocks.
+	b, _ := cst.miner.FindBlock()
+	err = cst.cs.AcceptBlock(b)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// Sleep to wait for possible calls to Broadcast to complete. We cannot
+	// wait on a channel because we don't know how many times broadcast has
+	// been called.
+	time.Sleep(1)
```

Comment:
Sleep is a bad way to do this. Also, I think it's worth noting that `time.Sleep(1)` will sleep for 1 nanosecond.

I don't think the consensus set should be calling `Broadcast` in a gothread. I think it makes sense to let the gateway handle the calls, and be able to block the consensus set or the transaction pool if it's backlogged.

Otherwise, you can either wait until Broadcast has been called at least once, but it's difficult to provide any sort of guarantee that it will be called exactly once. That sort of guarantee couldn't come from unit testing, it'd need to come from static code analysis.